src/event-handler/index.ts

```diff
@@ -1,13 +1,35 @@
+import type {
+  EmitterAnyEvent,
+  EmitterEventName,
+  EmitterWebhookEvent,
+  HandlerFunction,
+  Options,
+  State,
+  WebhookEventHandlerError,
+} from "../types";
 import {
   receiverOn as on,
   receiverOnAny as onAny,
   receiverOnError as onError,
 } from "./on";
 import { receiverHandle as receive } from "./receive";
 import { removeListener } from "./remove-listener";
-import { Options, State } from "../types";
 
-export function createEventHandler(options: Options<any>) {
+interface EventHandler<TTransformed = unknown> {
+  on<E extends EmitterEventName>(
+    event: E | E[],
+    callback: HandlerFunction<E, TTransformed>
+  ): void;
+  onAny(handler: (event: EmitterAnyEvent) => any): void;
+  onError(handler: (event: WebhookEventHandlerError) => any): void;
+  removeListener<E extends EmitterEventName>(
+    event: E | E[],
+    callback: HandlerFunction<E, TTransformed>
+  ): void;
+  receive(event: EmitterWebhookEvent): Promise<void>;
```

Comment:
Should this be equivalent to this?
https://github.com/octokit/webhooks.js/blob/d3df9f3d787990b60528cce5fba8365cdaa9ce61/src/index.ts#L37-L41

Reply:
Actually it should be the other way around - I've deliberately not modified that type because I don't want to blow out the scope of the PR more than needed, and the types are compatible so it's not a required change.

Reply:
> Actually it should be the other way around - I've deliberately not modified that type because I don't want to blow out the scope of the PR more than needed, and the types are compatible so it's not a required change.
👍🏽 

I will create an issue + PR for this if you are ok with it :)

Reply:
> I will create an issue + PR for this if you are ok with it :)

Go for gold! There's a number of potential improvements to the types and structure that I think can be made but have held off doing until #406 was landed as they're about smoothing out the internal types rather than external, and so would have been moot to land before a big refactor to the types.

That's also why I aimed for functionality first in that PR, rather than ensuring the types were completely perfect (i.e making sure there were no `type` aliases for the same type)